docs/src/languages/nodejs/debug.md

```diff
@@ -3,121 +3,244 @@ title: "Debugging"
-{{< /note >}}
+You can choose to debug in an environment deployed to Platform.sh
+or with your app running locally but connected to deployed services.
+In either case, make sure to debug in a non-production environment.
 
 {{% troubleshoot %}}
 
-## Remote debugging
+## 1. Create a new environment
 
-### Create a new environment
+Start by creating a new environment completely isolated from production but with the same data for debugging:
 
-```sh
-$ platform branch debug-branch
+```bash
+platform branch debug-branch
 ```
 
-### SSH into your app container
+## 2. Get access
 
-```sh
-$ platform ssh
+{{< codetabs >}}
+
+---
+title=Remote
+file=none
+highlight=false
+---
+
+Access your app container via [SSH](../../development/ssh/_index.md):
+
+```bash
+platform ssh
 ```
 
-In a multi-application project, you will be prompted to chose which application to SSH into. You can also specify the application you want using the `--app` option:
+<--->
+
+---
+title=Local
+file=none
+highlight=false
+---
 
-```sh
-$ platform ssh --app=my-app-name
+To access deployed apps and services, open tunnels to everything your app has relationships with:
+
+```bash
+platform tunnel:open
+```
+
+In the same terminal, set the relevant environment variables:
+
+```bash
+export PLATFORM_RELATIONSHIPS="$(platform tunnel:info --encode)"
+export PORT=8888
 ```
+<!--vale off -->
+{{< /codetabs >}}
+<!--vale on -->
 
-### Restart node in inspect mode 
 
-Once connected to your app container, run:
+## 3. Run your app in inspect mode
 
-```sh
-$ sv stop <app_name>
-$ node --inspect <js_file>
+{{< codetabs >}}
+
+---
+title=Remote
+file=none
+highlight=false
+---
+
+Stop the current process and restart it in inspect mode:
+
+```bash
+sv stop app
+node --inspect <START_FILE>
 ```
 
-`app_name` being the name of your application as defined in your `.platform.app.yaml` and `js_file` being your application entrypoint.
+Replace `<START_FILE>` with the file defined for [your app's `start` command](./_index.md#4-start-your-app).
 
-You should see something like:
+You get output something like this:
 
-```sh
-web@app.0:~$ node --inspect index.js 
+```bash
 Debugger listening on ws://127.0.0.1:9229/10701e5d-d627-4180-a967-d47a924c93c0
 For help, see: https://nodejs.org/en/docs/inspector
 Listening on port 8888
 ```
 
-### Forward the debugger port locally
+<--->
 
-In another terminal, create an SSH tunnel that forwards the 9229 port:
+---
+title=Local
+file=none
+highlight=false
+---
 
-```sh
-$ ssh -N -L9229:127.0.0.1:9229  $(platform  ssh --pipe)
+In the same terminal as the previous step, run the following command:
+
+```bash
+node --inspect <START_FILE>
 ```
 
-### Connect the debugger
+Replace `<START_FILE>` with the file defined for [your app's `start` command](./_index.md#usage-example).
+<!--vale off -->
+{{< /codetabs >}}
+<!--vale on -->
 
-You can now connect the debugger as if you were debugging a local application.
+{{< codetabs >}}
+
```

Comment:
:lady_beetle:
[The part below](https://pr-2360-k3ydbya-652soceglkw4u.eu-3.platformsh.site/languages/nodejs/debug.html#3-run-your-app-in-inspect-mode) (lines 112-155) is duplicated.